pdns/dnsdistdist/docs/rules-actions.rst

```diff
@@ -498,6 +498,18 @@ These ``DNSRule``\ s be one of the following items:
   :param KeyValueStore kvs: The key value store to query
   :param KeyValueLookupKey lookupKey: The key to use for the lookup
 
```

Comment:
I was wondering if there are certain illegal combinations of global rules/action and per-thread rules/actions. If so, that should be documented and detected (if possible).

Thinking about it some more, I think any combinations is acceptable, but you get runtime (or silent?) errors if you try to access the global context from a per-thread rule or action. Some explanation or example describing what's in the global context (or more importantly what's not in the per-thread context) would be helpful:

Just an example: `DNSClass.IN`, is that a name I can use in a per-thread context?

Reply:
Pretty much everything is allowed, as long as you remember that you are in a separate Lua context and not in the global one, as documented below. All constants are available in the per-thread context, and all FFI functions. Objects and their bindings that are not usable in a FFI context (`DNSQuestion`, `DNSDistProtoBufMessage`, `PacketCache`, ...) are not available.
That seems quite obvious to me but I'll add a couple lines to the descriptions.